### PR TITLE
SEM-9192-Solarwinds-changes-JAXB

### PR DESCRIPTION
**Pull request by michal-friedl-swi**
Changes were gotten by decompilation of this plugin from solarwinds artifactory

### CODE REVIEW

File: src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java [hunk 1]
```diff
@@ -147,7 +153,37 @@ private void fix() throws MojoExecutionException
...
+        String inContent = new String(Files.readAllBytes(file.toPath()), charset);
+        String outContent = this.sortXmlSeeAlso(inContent);
+        if (!inContent.equals(outContent))
+        {
+            Files.write(file.toPath(), outContent.getBytes(charset), new OpenOption[0]);
+        }
+    }
+    private String sortXmlSeeAlso(String inContent)
+    {
+        Pattern p = Pattern.compile("(@XmlSeeAlso\\s*\\(\\s*\\{)([^})]*)(}\\s*\\))", 8);
```
Comment (libor-sasak-swi) [inline src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java:168]: replace 8 with Pattern.MULTILINE

File: src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java [hunk 2]
```diff
@@ -147,7 +153,37 @@ private void fix() throws MojoExecutionException
...
+        Matcher m = p.matcher(inContent);
+        if (m.find())
+        {
+            String items = (String) Arrays.stream(m.group(2).trim().split(",")).map((x) ->
+            {
+                return System.lineSeparator() + "     " + x.trim();
+            }).sorted().collect(Collectors.joining(","));
```
Comment (libor-sasak-swi) [inline src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java:175]:
```suggestion
            String items = (String) Arrays.stream(m.group(2).trim().split(",")).map((x) ->
					System.lineSeparator() + "     " + x.trim()).sorted().collect(Collectors.joining(","));
```

File: src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java (hunk 1 above)
Comment (michal-friedl-swi) [reply src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java]: done

File: src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java [hunk 3]
```diff
@@ -147,7 +153,37 @@ private void fix() throws MojoExecutionException
...
     }
-    
+
+    private void sortXmlSeeAlsoInFile(File file, Charset charset) throws IOException
+    {
+        String inContent = new String(Files.readAllBytes(file.toPath()), charset);
+        String outContent = this.sortXmlSeeAlso(inContent);
+        if (!inContent.equals(outContent))
+        {
+            Files.write(file.toPath(), outContent.getBytes(charset), new OpenOption[0]);
```
[the remark this answers is not in the document]
Comment (michal-friedl-swi) [reply src/main/java/io/github/zlika/reproducible/StripJaxbMojo.java]: done

**Review by martin-kolesar-swi — APPROVED** [review]

**Pull request merged**